src/Servers/IIS/IIS/test/Common.FunctionalTests/ShadowCopyTests.cs

```diff
@@ -0,0 +1,216 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Net.Sockets;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Server.IIS.FunctionalTests.Utilities;
+using Microsoft.AspNetCore.Server.IIS.FunctionalTests;
+using Microsoft.AspNetCore.Server.IntegrationTesting;
+using Microsoft.AspNetCore.Testing;
+using Microsoft.AspNetCore.Testing.xunit;
+using Microsoft.Extensions.Logging;
+using Xunit;
+using Microsoft.AspNetCore.Server.IntegrationTesting.IIS;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.Server.IIS.FunctionalTests
+{
+    [Collection(PublishedSitesCollection.Name)]
+    public class ShadowCopyTests : IISFunctionalTestBase
+    {
+        public ShadowCopyTests(PublishedSitesFixture fixture) : base(fixture)
+        {
+        }
+
+        // TODO check if app init module works here
+        // TODO verify app init on azure app services.
+        // TODO check out of proc?
+        // TODO check startup failure
+        // TODO check shutting down?
+
+        [ConditionalFact]
+        public async Task ShadowCopyWorks()
+        {
+            var directory = CreateTempDirectory();
+            var deploymentParameters = Fixture.GetBaseDeploymentParameters();
+            deploymentParameters.HandlerSettings["enableShadowCopy"] = "true";
+            deploymentParameters.HandlerSettings["shadowCopyDirectory"] = directory.FullName;
+
+            var deploymentResult = await DeployAsync(deploymentParameters);
+            await deploymentResult.HttpClient.GetStringAsync("Wow!");
```

Comment:
Have the app return the loaded assembly disk location to make sure it matches the shadow dir? So far this test is only confirming the app didn't crash. We can't tell if it used shadow copy or not.